src/Features/CSharp/Portable/Completion/KeywordRecommenders/SwitchKeywordRecommender.cs

```diff
@@ -16,7 +17,35 @@ protected override bool IsValidContext(int position, CSharpSyntaxContext context
         {
             return
                 context.IsStatementContext ||
-                context.IsGlobalStatementContext;
+                context.IsGlobalStatementContext ||
+                IsAfterExpression(context);
+        }
+
+        // `switch` follows expressions in switch expressions
+        internal static bool IsAfterExpression(CSharpSyntaxContext context)
+        {
+            var token = context.TargetToken;
+            if (token.Parent == null)
+            {
+                return false;
+            }
+
+            var ancestors = token.Parent.AncestorsAndSelf();
+
+            foreach (var ancestor in ancestors)
+            {
+                if (ancestor is ExpressionSyntax)
+                {
+                    return true;
+                }
+
+                if (ancestor is LambdaExpressionSyntax || ancestor is StatementSyntax)
+                {
+                    break;
+                }
+            }
+
+            return false;
```

Comment:
can this reuse the same logic that `is` and `as` keywords use? those also show up generally after an expression

Reply:
Excellent idea. Thanks